packages/components/src/navigation/test/index.js

```diff
@@ -45,92 +51,291 @@ const testNavigation = ( { activeItem, rootTitle, showBadge } = {} ) => (
 	</Navigation>
 );
 
+const TestNavigationControlled = () => {
+	const [ activeItem, setActiveItem ] = useState( 'item-1' );
+	const [ activeMenu, setActiveMenu ] = useState( 'root' );
+
+	// Mock navigation link.
+	const MockLink = ( { href, children } ) => (
+		<button
+			href={ href }
+			// Since we're not actually navigating pages, simulate it with on onClick.
```

Comment:
Nit: Potential typo using 'on' instead of 'an'

Reply:
Good spot! Updated it in both the tests, and in the StoryBook example (where I "borrowed" the code from)

https://github.com/WordPress/gutenberg/pull/41668/commits/88a8fb7fb53f9589ce54b4207e3705b8b832549b